app/src/main/java/com/simprints/intentlauncher/ui/events/ResponseEventsScreen.kt

```diff
@@ -0,0 +1,505 @@
+import com.simprints.intentlauncher.ui.events.model.EventSortingOption
+import com.simprints.intentlauncher.ui.intent.model.ResponseEvent
+import com.simprints.intentlauncher.ui.theme.AppColorScheme
+import com.simprints.intentlauncher.ui.theme.AppTypography
+import kotlinx.coroutines.launch
+import java.util.Date
+
+private val eventMock = ResponseEvent(
+    id = "738299cd2-389f-a9ce-3d11a3948ce",
+    type = "MOCKED_EVENT",
+    scopeId = "scope id",
+    projectId = "project id",
+    createdAtMs = 0,
+    payload = mapOf(
+        "key 1" to "value 1",
+        "key 2" to listOf("list value one", "list value two"),
+        "key 3" to mapOf(
+            "nested key 1" to "nested value 1",
+            "nested key 2" to mapOf(
+                "second nested key 1" to "second nested value 1 with very long sentence",
+                "second nested key 2" to "second nested value 2",
+            )
+        ),
+        "key 4" to "value 4"
+    )
+)
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun ResponseEventsScreen(
+    navController: NavHostController,
+    intentId: String
+) {
+    val window = (LocalContext.current as? Activity)?.window
+    val clipboardManager = LocalClipboardManager.current
+    val vm =
+        hiltViewModel<ResponseEventsScreenViewModel, ResponseEventsScreenViewModel.Factory> { factory ->
+            factory.create(
+                intentId = intentId,
+                initialSoftInputMode = window?.attributes?.softInputMode?.takeUnless { it == 0 }
+                    ?: WindowManager.LayoutParams.SOFT_INPUT_ADJUST_PAN
+            )
+        }
+    val viewState by vm.state.collectAsStateLifecycleAware(initial = vm.initialViewState)
+    // What kind of dialog is displayed
+    var eventDialogType by remember { mutableStateOf<EventDialogType?>(null) }
+    val scope = rememberCoroutineScope()
+    val modalSheetState = rememberModalBottomSheetState(
+        initialValue = ModalBottomSheetValue.Hidden,
+        confirmStateChange = { it != ModalBottomSheetValue.HalfExpanded },
+        skipHalfExpanded = true,
+    )
+    val toastText = remember { mutableStateOf("") }
+    ToastLauncher(toastText)
+    if (modalSheetState.currentValue == ModalBottomSheetValue.Hidden) {
+        eventDialogType = null
+    }
+
+    // Setting SoftInputMode to SOFT_INPUT_ADJUST_RESIZE so that content won't shift away from the
+    // screen when the keyboard is displayed. Setting it back to the initial value once the widget
+    // is no longer visible
+    window?.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_RESIZE)
+    DisposableEffect(Unit) {
+        onDispose {
+            window?.setSoftInputMode(vm.initialSoftInputMode)
+        }
+    }
+    ModalBottomSheetLayout(
+        sheetState = modalSheetState,
+        sheetShape = RoundedCornerShape(topStart = 12.dp, topEnd = 12.dp),
+        sheetContent = {
+            when (val dialogType = eventDialogType) {
```

Comment:
Either the whole `when` or the branches could be separate composables for easier comprehension.